src/components/TemplateLibrary/index.js

```diff
@@ -200,24 +203,44 @@ const LibraryComponent = () => {
           const customXmlPart = result.value[0];
           customXmlPart.getNodesAsync('*/*', result => {
             if (result.status === Office.AsyncResultStatus.Succeeded) {
+              let identifierExists = false;
               let newXml = XML_HEADER + `<templates xmlns="${CUSTOM_XML_NAMESPACE}">`;
               if (result.value.length > 0) {
                 for (let node=0; node < result.value.length; ++node) {
                   if (result.value[node].namespaceUri !== templateIdentifier) {
                     newXml += `<template xmlns="${result.value[node].namespaceUri}" />`;
                   }
+                  if(result.value[node].namespaceUri === templateIdentifier){
+                    identifierExists = true;
+                  }
                 }
               }
-              newXml += `<template xmlns="${templateIdentifier}" />`;
-              newXml += '</templates>';
-              Office.context.document.customXmlParts.getByNamespaceAsync(CUSTOM_XML_NAMESPACE, res => {
-                if (res.status === Office.AsyncResultStatus.Succeeded) {
-                  for (let index=0; index<res.value.length; ++index) {
-                    res.value[index].deleteAsync();
+              if(!identifierExists){
+                setup(ciceroMark, template);
+                newXml += `<template xmlns="${templateIdentifier}" />`;
+                newXml += '</templates>';
+                Office.context.document.customXmlParts.getByNamespaceAsync(CUSTOM_XML_NAMESPACE, res => {
+                  if (res.status === Office.AsyncResultStatus.Succeeded) {
+                    for (let index=0; index<res.value.length; ++index) {
+                      res.value[index].deleteAsync();
+                    }
                   }
-                }
-              });
-              Office.context.document.customXmlParts.addAsync(newXml);
+                });
+                Office.context.document.customXmlParts.addAsync(newXml);
+              }else{
+                toast(
+                  {
+                    title: 'Duplicate template',
+                    description: <p>Template cannot be inserted as it is already present in the document.</p>,
+                    type: 'error',
+                    time: 5000,
+                    animation: 'fly down',
+                  },
+                  () => console.log('toast closed'),
+                  () => console.log('toast clicked'),
+                  () => console.log('toast dismissed')
```

Comment:
These `console` statements are not required as it doesn't matter to us how the user interacts with the toast. You can remove them.